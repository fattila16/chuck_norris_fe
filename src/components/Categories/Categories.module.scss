@import 'src/assets/scss/color-palette';
@import 'src/assets/scss/vars';

.container {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.ul {
    display: grid;
    grid-gap: 1rem;
    grid-template-columns: repeat(3, minmax(150px, 1fr));
    margin: 0;
    padding: 0;
    
    button {
        display: grid;
        place-items: center;
        list-style-type: none;
        height: 80px;
        background-color: map-get($active-theme , liver-dogs );
        color: map-get($active-theme , silk);
        border-radius: 20px;
        cursor: pointer;
    }
}

@media only screen and (max-width: $mobile-breakpoint) {
    .ul {
        margin: 0;
        padding: 0;
        grid-template-columns: repeat(2, minmax(80px, 1fr));
        button {
            height: 50px;
            width: 30vw;
        }
        margin-bottom: 40px;
    }
}