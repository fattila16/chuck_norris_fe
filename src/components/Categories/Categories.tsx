import React from 'react';
import css from './Categories.module.scss';
import Button from '../Button/Button';

interface CategoriesProps {
    categories: string[];
    handleCategoryChange: (category: string) => void;
    error?: Error;
}
const Categories: React.FC<CategoriesProps> = (props: CategoriesProps) => {
    const { categories, error, handleCategoryChange } = props;
    if (error) return <>Error</>;
    return (
        <section className={css.container}>
            <h3>Categories List:</h3>
            <ul className={css.ul}>
                {categories?.map((val: any) => (
                    <Button
                        id={val}
                        key={val}
                        onClick={() => {
                            handleCategoryChange(val);
                        }}
                        title={val}
                        size="medium"
                    />
                ))}
            </ul>
        </section>
    );
};

export default Categories;
