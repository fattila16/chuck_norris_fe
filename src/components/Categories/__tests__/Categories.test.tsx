import React from 'react';
import { shallow, mount } from 'enzyme';
import Categories from '../Categories';

const mockHandleChangeCategory = jest.fn();

describe('[Categories]', () => {
    it('Should render Categories component without error', () => {
        const component = shallow(
            <Categories categories={['animal', 'fashion']} handleCategoryChange={mockHandleChangeCategory} />,
        );
        expect(component.getElements()).toMatchSnapshot();
    });

    it('Should call handleChangeCategory when category button is clicked', () => {
        const component = mount(
            <Categories categories={['animal', 'fashion']} handleCategoryChange={mockHandleChangeCategory} />,
        );
        component.find('button[id="animal"]').simulate('click');
        expect(mockHandleChangeCategory).toBeCalled();
    });

    it('Should render Error if error prop is present', () => {
        const component = shallow(<Categories categories={[]} error={new Error()} handleCategoryChange={jest.fn()} />);
        expect(component.text()).toMatch('Error');
    });
});
