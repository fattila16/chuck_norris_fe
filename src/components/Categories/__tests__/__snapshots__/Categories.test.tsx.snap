// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Categories] Should render Categories component without error 1`] = `
Array [
  <section
    className="container"
  >
    <h3>
      Categories List:
    </h3>
    <ul
      className="ul"
    >
      <Button
        id="animal"
        onClick={[Function]}
        size="medium"
        title="animal"
      />
      <Button
        id="fashion"
        onClick={[Function]}
        size="medium"
        title="fashion"
      />
    </ul>
  </section>,
]
`;
